File: algorithms/python/Challenge_Cheapest_Flights_Within_K_Stops.py
class Solution:
    def findCheapestPrice(self, n: int, flights: List[List[int]], src: int, dst: int, K: int) -> int:
        cost = [100000 for _ in range(n)]
        cost[src] = 0
        
        for i in range(K+1):
            tmp = cost[:]
            for flight in flights:
                cur, nex, pri = flight
                
                if cost[cur] == 100000:
                    continue
                
                if tmp[nex] > cost[cur] + pri:
                    tmp[nex] = cost[cur] + pri
            
            cost = tmp
        
        return cost[dst] if not cost[dst] == 100000 else -1

File: algorithms/java/Jun.2020.challenge/Challenge_Cheapest_Flights_Within_K_Stops.java
class Solution {
    
    public int findCheapestPrice(int n, int[][] flights, int src, int dst, int K) {
        
        int[] cost = new int[n];
        Arrays.fill(cost, Integer.MAX_VALUE);
        cost[src] = 0;
        
        for(int k = 0 ; k <= K ; k++) {
            int[] tmp = Arrays.copyOf(cost, n);
            for(int[] flight : flights) {
                int cur = flight[0], next = flight[1], price = flight[2];
                
                if(cost[cur] == Integer.MAX_VALUE)
                    continue;
                
                if(tmp[next] > cost[cur]+price) {
                    tmp[next] = cost[cur]+price;
                }
            }
            cost = tmp;
        }
        
        return cost[dst] == Integer.MAX_VALUE ? -1 : cost[dst];
    }
}
